dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -345,13 +360,31 @@ def _update_categories(
         :type df_series: pandas.DataFrame
         :return: None
         """
-        category_count = df_series.value_counts(dropna=False).to_dict()
```

Comment:
I think something like this and it allows to reuse for `__add__` as well.

```python
def _get_categories(....):


def _merge_categories(...):


def _update_categories(...):

    # new_summary = _get_categories(new_data)
    cms = count_min_sketch(self..., self...)
    heavy_hitter_dict = {}
    for i in df_series:
        cms.update(i)
        # approximate heavy-hitters
        if cms.get_estimate(i) >= int(
            len(df_series) / self._heavy_hitters_threshold
        ):
            heavy_hitter_dict[i] = self.cms.get_estimate(i)

    
    
    # old_summary = self.....of info
    # _merge_categories(new_summary, old_summary)
    self.cms = self.cms + cms
    
    categories = utils.add_nested_dictionaries(
        self._categories, heavy_hitter_dict
    )
    
    total_samples = len(data) + self.total_samples
    for cat in categories:
        if categories["cat"] < (total_samples / self._heavy_hitters_threshold):
            categories.pop("cat")
            
    self._categories = categories
```

Reply:
if we abstract to get and merge, `__add__` can utilize the merge if we do it via a new and prev

Reply:
Updated above for new thoughts wrt secondary list to cover edge cases.